api/src/services/monitoring.js

```diff
@@ -146,26 +147,98 @@ const getFeedbackCount = () => {
     });
 };
 
+const STATUS_KEYS = ['due', 'scheduled', 'muted', 'failed', 'delivered'];
+const fromEntries = (keys, value) => {
+  // "shim" of Object.fromEntries
+  const result = {};
+  keys.forEach(key => result[key] = value);
+  return result;
+};
+
 const getOutgoingMessageStatusCounts = () => {
   return db.medic.query('medic-admin/message_queue', { reduce: true, group_level: 1 })
     .then(counts => {
-      const result = {
-        due: 0,
-        scheduled: 0,
-        muted: 0
-      };
+      const result = fromEntries(STATUS_KEYS, 0);
       counts.rows.forEach(row => {
         result[row.key[0]] = row.value;
       });
       return result;
     })
     .catch(err => {
       logger.error('Error fetching outgoing message status count: %o', err);
-      return {
-        due: -1,
-        scheduled: -1,
-        muted: -1
-      };
+      return fromEntries(STATUS_KEYS, -1);
+    });
+};
+
+const getWeeklyOutgoingMessageStatusCounts = () => {
+  const startDate = moment().startOf('day').subtract(7, 'days').valueOf();
+  const endDate = moment().valueOf();
+
+  const options = STATUS_KEYS.map(key => ({
+    start_key: [key, startDate],
+    end_key: [key, endDate],
+    reduce: true,
+  }));
+
+  const query = (options) => db.medic
+    .query('medic-admin/message_queue', options)
+    .catch(err => {
+      logger.error(`Error fetching weekly outgoing message status counts ${options.start_key}: %o`, err);
+      return { rows: [{ value: -1 }] };
+    });
+
+  return Promise
+    .all(options.map(options => query(options)))
+    .then(counts => {
+      const result = fromEntries(STATUS_KEYS, 0);
+      counts.forEach((count, idx) => {
+        const row = count && count.rows && count.rows[0];
+        if (row) {
+          result[STATUS_KEYS[idx]] = row.value;
+        }
+      });
+
+      return result;
+    });
+};
+
+const getLastHundredStatusCountsPerGroup = ({group, statuses}) => {
+  const options = {
+    descending: true,
+    start_key: [group, moment().valueOf()],
+    endkey: [group, 0],
+    limit: 100,
+  };
+
+  return db.medic
+    .query('medic-sms/messages_by_last_updated_state', options)
+    .then(results => {
+      const counts = fromEntries(statuses, 0);
+      results.rows.forEach(row => {
+        const status = row.key[2];
+        counts[status] = counts[status] || 0;
+        counts[status]++;
+      });
+
+      return counts;
+    })
+    .catch(err => {
+      logger.error(`Error fetching last 100 final status messages: %o`, err);
+      return fromEntries(statuses, -1);
+    });
+};
+
+const getLastHundretStatusUpdatesCounts = () => {
+  const groups = [
```

Comment:
This feels like it should be coming from one source, I mean not having the declare these constants about `forwarded-to-gateway`, `received-by-gateway1, etc and the group it belongs to, in many places. 
Do you think we can make a single source?

Reply:
I'm not sure about this, it would require a major overhaul of many services. 
But these "groups" are only used here. The statuses themselves are used in other places. 

Maybe the middle group would be to make constants out of each status and export those, but still use statuses independently. 
I'll create an issue.